src/connectors/impls/clickhouse.rs

```diff
@@ -0,0 +1,290 @@
+// Copyright 2021, The Tremor Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod conversion;
+
+use std::fmt::{self, Display, Formatter};
+
+use crate::connectors::{prelude::*, utils::url::ClickHouseDefaults};
+
+use clickhouse_rs::{
+    types::{DateTimeType, SqlType},
+    Block, Pool,
+};
+use either::Either;
+
+#[derive(Default, Debug)]
+pub(crate) struct Builder {}
+
+#[async_trait::async_trait]
+impl ConnectorBuilder for Builder {
+    fn connector_type(&self) -> ConnectorType {
+        "clickhouse".into()
+    }
+
+    async fn build(&self, alias: &str, raw_config: &ConnectorConfig) -> Result<Box<dyn Connector>> {
+        let config = raw_config
+            .config
+            .as_ref()
+            .ok_or_else(|| ErrorKind::MissingConfiguration(alias.to_string()))?;
+
+        let config = ClickhouseConfig::new(config)?;
+
+        Ok(Box::new(Clickhouse { config }))
+    }
+}
+
+pub struct Clickhouse {
+    config: ClickhouseConfig,
+}
+
+#[async_trait::async_trait]
+impl Connector for Clickhouse {
+    async fn create_sink(
+        &mut self,
+        sink_context: SinkContext,
+        builder: SinkManagerBuilder,
+    ) -> Result<Option<SinkAddr>> {
+        let db_url = self.connection_url();
+        let columns = self
+            .config
+            .columns
+            .iter()
+            .map(|Column { name, type_ }| (name.clone(), type_.clone()))
+            .collect();
+
+        let sink = ClickhouseSink {
+            db_url,
+            pool: None,
+            columns,
+        };
+        builder.spawn(sink, sink_context).map(Some)
+    }
+
+    fn codec_requirements(&self) -> CodecReq {
+        CodecReq::Structured
+    }
+}
+
+impl Clickhouse {
+    fn connection_url(&self) -> String {
+        let url = self.config.url.as_str();
+        let compression = self.config.compression.unwrap_or_default();
+
+        format!("{url}?compression={compression}")
+    }
+}
+
+#[derive(Deserialize)]
+struct ClickhouseConfig {
+    url: Url<ClickHouseDefaults>,
+    compression: Option<Compression>,
+    columns: Vec<Column>,
+}
```

Comment:
the table name is actually still missing here :)

Reply:
Nice catch! This should be fixed by [1353320](https://github.com/tremor-rs/tremor-runtime/pull/1538/commits/135332017d87623f6d3b41296522d9518ad0ac70)